src/app/features/automated-status-update/automated-field-mapping/automated-field-mapping.component.ts

```diff
@@ -176,12 +176,17 @@ export class AutomatedFieldMappingComponent implements OnInit {
     });
     if (this.isInvalid) return;
 
-    const formattedMappings: { [key: string]: any } = {};
-    Object.entries(this.selectedMappings).forEach(([key, value]) => {
-      formattedMappings[key] = this.schemaService.valueToDatabaseFormat(
-        value,
-        this.targetFieldConfig,
-      );
+    const formattedMappings: { [key: string]: any } = {
+      ...this.selectedMappings,
+    };
+    Object.entries(this.mappingForms).forEach(([key, mappingForm]) => {
+      const value = mappingForm.formGroup.get(this.targetFieldConfig.id)?.value;
+      if (value) {
```

Comment:
I thought this might prevent it from saving a deleted mapped value. That doesn't seem to be a problem ... But do we really need the check here or can we just always use whatever value is in the form control currently?